src/QMCWaveFunctions/lcao/LCAOrbitalBuilder.cpp

```diff
@@ -806,14 +831,26 @@ void readRealMatrixFromH5(hdf_archive& hin, const std::string& setname, Matrix<L
 void LCAOrbitalBuilder::LoadFullCoefsFromH5(hdf_archive& hin,
                                             int setVal,
                                             PosType& SuperTwist,
-                                            Matrix<std::complex<RealType>>& Ctemp)
+                                            Matrix<std::complex<RealType>>& Ctemp,
+                                            bool MultiDet)
 {
   Matrix<RealType> Creal(Ctemp.rows(), Ctemp.cols());
   Matrix<RealType> Ccmplx(Ctemp.rows(), Ctemp.cols());
 
   char name[72];
   std::string setname;
-  sprintf(name, "%s%d", "/Super_Twist/eigenset_", setVal);
+  ///When running Single Determinant calculations, MO coeff loaded based on occupation and lowest eingenvalue.
+  ///However, for solids with multideterminants, orbitals are order by kpoints; first all MOs for kpoint 1, then 2 etc
+  /// The multideterminants occupation is specified in the input/HDF5 and theefore as long as there is consistency between
+  /// the order in which we read the orbitals and the occupation, we are safe. In the case of Multideterminants generated
+  /// by pyscf and Quantum Package, They are stored in the same order as generated for quantum package and one should use
+  /// the orbitals labled eigenset_unsorted.
+
+  if (MultiDet == false)
+    sprintf(name, "%s%d", "/Super_Twist/eigenset_", setVal);
+  else
```

Comment:
Someday, both routes should use the same dataset (the one ordered by k-points).  I believe this is what prevents straightforward support for single determinant excitations (apart from the einspline plumbing of course).

Reply:
Agreed... Some thinking on how to properly do this should be done. For now, I do not have enough cases to see what we will be using more in the future... Reading the same source requires reading also the eingenvalues (to do the sorting in QMCPACK). As always, I wanted to have a working version and let people decide on a working code instead of speculative code. Since this is needed for BES paper, it was an extra motivation to push both and let the future decide wich route to take (for more on this point see next point raised by @jtkrogel )